resources/hab_build.rb

```diff
@@ -20,20 +22,28 @@
     command "sudo -E #{hab_binary} studio" \
             " -r #{hab_studio_path}" \
             " build #{plan_dir}"
-    environment('TERM' => 'vt100',
-                'HAB_ORIGIN' => origin,
-                'HAB_NONINTERACTIVE' => 'true')
+    environment({ 'TERM' => 'vt100',
+                  'HAB_ORIGIN' => origin,
+                  'HAB_NONINTERACTIVE' => 'true' }.merge(
+                    new_resource.environment
```

Comment:
But doesn't that it will stomp on those if I pass them in? 

If it set these as defaults, which is what I had it do, that wouldn't surprise me as a user. If it ended up setting it to be the opposite of what I passed in that would not be what I expected.

People won't normally be messing with these, but it seems what I have there now is closer to what we want.